x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/analytics/AnalyticsTemplateRegistry.java

```diff
@@ -29,25 +29,34 @@
 import static org.elasticsearch.xpack.core.ClientHelper.ENT_SEARCH_ORIGIN;
 
 public class AnalyticsTemplateRegistry extends IndexTemplateRegistry {
-    public static final String ROOT_RESOURCE_PATH = "/org/elasticsearch/xpack/entsearch/analytics/";
+
+    // Data stream related constants.
+    public static final String EVENT_DATA_STREAM_DATASET = "events";
+    public static final String EVENT_DATA_STREAM_TYPE = "behavioral_analytics";
+    public static final String EVENT_DATA_STREAM_INDEX_PREFIX = EVENT_DATA_STREAM_TYPE + "-" + EVENT_DATA_STREAM_DATASET + "-";
+    public static final String EVENT_DATA_STREAM_INDEX_PATTERN = EVENT_DATA_STREAM_INDEX_PREFIX + "*";
+
+    // Resource config.
+    protected static final String ROOT_RESOURCE_PATH = "/org/elasticsearch/xpack/entsearch/analytics/";
 
     // This number must be incremented when we make changes to built-in templates.
-    public static final int REGISTRY_VERSION = 1;
+    protected static final int REGISTRY_VERSION = 1;
```

Comment:
Why `protected`? Are we expecting subclasses for `AnalyticsTemplateRegistry`?

Reply:
I have another PR with a class in the same package that use this variable.
I will probably push those into a file that hold all these constant.

Reply:
> I have another PR with a class in the same package that use this variable.

Maybe use default package visibility then? Sorry for nagging, I see `protected` as a strong signal that it is used in a subclass. Let's use default package visibility if that's the use case 👍